### PR TITLE
Minimal way to make ReduceLROnPlateau composable with SequentialLR and ChainedScheduler

### LINKED ISSUE #68978: SequentialLR cannot be used with ReduceLROnPlateau due to .step() not allowing for optional arguments
## 🐛 Bug

Currently, SequentialLR can only be used with schedulers that inherit from _LRScheduler or in other words adhere to the way .step() is called, i.e. without any arguments. This is not the case for the built in ReduceLROnPlateau as it takes in a metric value. Therefore, when calling SequentialLR.step() without arguments, ReduceLROnPlateau will raise an error once its milestone is reached. However, when calling SequentialLR.step(metric), SequentialLR will raise an error due to too many arguments.

Also see https://github.com/PyTorchLightning/pytorch-lightning/issues/10759

## To Reproduce
```python
scheduler1 = ConstantLR(self.opt, factor=0.1, total_iters=2)
scheduler2 = ReduceLROnPlateau (self.opt, gamma=0.9)
scheduler = SequentialLR(self.opt, schedulers=[scheduler1, scheduler2], milestones=[2])
for epoch in range(100):
   train(...)
   validate(...)
   scheduler.step()  # raises error after epoch>=2
   scheduler.step(metric)  # raises error immediately
```

## Expected behavior
SequentialLR should allow for arbitrary arguments in step to be used with the underlying scheduler. Either the user takes care of passing the right argument at the right time or there is a built in mechanic (if that is even possible).

### LINKED ISSUE #110761: Using ChainedScheduler with ReduceLROnPlateau leads to unexpected keyword argument error
### 🚀 The feature, motivation and pitch

I want to use ReduceLROnPlateau together with another schedule, say a warm-up or cosine schedule.
My understanding is that the suggested way to do that is through ``ChainedScheduler``.
However, when I try to call ``step`` with a validation loss, I get a ``TypeError``, since ``ChainedScheduler`` doesn't accept ``kwargs`` in ``step``.
I'm still pretty new to pytorch, so please let me know if I'm overlooking some other solution.

I don't see a super obvious solution. One would be to pass all ``kwargs`` to all steps in the chain, but that would mean all the schedulers would have to accept arbitrary ``kwargs`` which would hide errors. The other option that comes to mind would be to use inspect to figure out which parameters to pass to which scheduler. Or the ``step`` of ``ChainedScheduler`` could take a dict mapping from schedulers to parameters, but that seems more complicated than just calling all the schedules explicitly and not using ``ChainedScheduler``.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pupeno**
Fixes #68978  
Fixes #110761

Video overview to try to save you some time and hassle reviewing: https://youtu.be/oHZofnGd_AQ

Fixes `ReduceLROnPlateau` composition with `SequentialLR` and `ChainedScheduler`.

Both composition schedulers previously rejected `ReduceLROnPlateau` because it requires a metric in `step()`. This change removes that restriction and adds an optional `metrics` argument to each composer’s `step()`, forwarding it only to schedulers that can consume it. Existing zero-argument calls remain unchanged.

To make `ReduceLROnPlateau` participate correctly in the existing scheduler-composition lifecycle, this also:

- aligns its initial-step state with `LRScheduler` conventions
- preserves `initial_lr` when it is constructed
- implements `get_lr()` as the current learning rates, allowing `SequentialLR`  milestone handoffs to work
- raises a clear error when a plateau scheduler is stepped without its required  metric

Nested `SequentialLR`/`ChainedScheduler` compositions forward metrics through their containers, so a plateau scheduler can be nested within either composition type.

You can find the sample code here: https://github.com/pupeno/pytorch-workspace/tree/main/195633_minimal_reduce_lr_on_plateau_composable

An alternative approach is available in #195634. The two PRs are mutually exclusive.

This PR depends on https://github.com/pytorch/pytorch/pull/196242

### CODE REVIEW

**Comment by pupeno** [comment]
I have signed the CLA, but I couldn't find a way to retrigger the check.
